src/View/Helper/DocBlockHelper.php

```diff
@@ -93,60 +85,41 @@ public function associatedEntityTypeToHintType($type, Association $association)
      * @param array $propertySchema The property schema to use for generating the type map.
      * @return array The property DocType map.
      */
-    public function buildEntityPropertyHintTypeMap(array $propertySchema)
+    public function buildEntityPropertyHints(array $propertySchema)
     {
         $properties = [];
-        foreach ($propertySchema as $property => $info) {
-            if ($info['kind'] === 'column') {
-                $type = $this->columnTypeToHintType($info['type']);
-                if (!empty($info['null'])) {
-                    $type .= '|null';
-                }
-
-                $properties[$property] = $type;
+        $columns = array_filter($propertySchema, function (array $info) {
+            return $info['kind'] === 'column';
+        });
+        foreach ($columns as $property => $info) {
+            $type = $this->columnTypeToHintType($info['type']);
+            if (!empty($info['null'])) {
+                $type .= '|null';
             }
+
+            $properties[] = [
+                'prefix' => '@property',
+                'type' => $type,
+                'property' => sprintf('$%s', $property),
+                'comment' => $info['comment'],
+            ];
         }
 
-        return $properties;
-    }
+        if (!empty($properties)) {
+            $properties[] = [];
+        }
 
-    /**
-     * Builds a map of entity associations as DocBlock types for use
-     * in generating `@property` hints.
-     *
-     * This method expects a property schema as generated by
-     * `\Bake\Shell\Task\ModelTask::getEntityPropertySchema()`.
-     *
-     * The generated map has the format of
-     *
-     * ```
-     * [
-     *     'property-name' => 'doc-block-type',
-     *     ...
-     * ]
-     * ```
-     *
-     * @see \Bake\Shell\Task\ModelTask::getEntityPropertySchema
-     *
-     * @param array $propertySchema The property schema to use for generating the type map.
-     * @return array The property DocType map.
-     */
-    public function buildEntityAssociationHintTypeMap(array $propertySchema)
```

Comment:
We can't safely remove these methods. The DocblockHelper is part of the public interface of bake as it could be used in userland templates.

Reply:
Or in https://github.com/dereuromark/cakephp-ide-helper :)

Reply:
Got it. I will think about another implementation.